ipamd/ipamd.go

```diff
@@ -161,10 +162,10 @@ type IPAMContext struct {
 	primaryIP            map[string]string
 	lastNodeIPPoolAction time.Time
 	lastDecreaseIPPool   time.Time
-
 	// reconcileCooldownCache keeps timestamps of the last time an IP address was unassigned from an ENI,
 	// so that we don't reconcile and add it back too quickly if IMDS lags behind reality.
 	reconcileCooldownCache ReconcileCooldownCache
+	terminating            *int32 // Flag to warn that the pod is about to shut down.
```

Comment:
you could just make this an int32 and then when doing the `atomic.StoreInt32()` call below, just refer to the address of this field, like so:

```go
atomic.StoreInt32(&c.terminating, 1)
```

Reply:
Good idea, makes it cleaner. (Was testing the concept first with a boolean, but that's not thread safe.)